cgroup2/utils.go

```diff
@@ -379,41 +389,55 @@ func systemdUnitFromPath(path string) string {
 func readHugeTlbStats(path string) []*stats.HugeTlbStat {
 	usage := []*stats.HugeTlbStat{}
 	keyUsage := make(map[string]*stats.HugeTlbStat)
+
 	f, err := os.Open(path)
 	if err != nil {
 		return usage
 	}
-	files, err := f.Readdir(-1)
+
+	files, err := f.Readdirnames(-1)
```

Comment:
You can change to improve further marginally by using f.ReadDir(-1).
See: https://go.dev/play/p/aMmQKCKBZpN

I got 
```
-timeout 30m

goos: linux
goarch: amd64
pkg: github.com/manugupt1/file
cpu: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
BenchmarkReaddir-8        	     871	   1300868 ns/op	  130113 B/op	    1439 allocs/op
BenchmarkReadDir-8        	    3076	    370488 ns/op	   48972 B/op	     957 allocs/op
BenchmarkReadDirNames-8   	    3139	    366333 ns/op	   48973 B/op	     957 allocs/op
PASS
	github.com/manugupt1/file	coverage: 50.0% of statements
ok  	github.com/manugupt1/file	3.655s
```

Reply:
Hmm, this is what I swapped to originally but found the latter shaved down a few more allocations. Let me try again

Reply:
Hmm! I am not sure why; maybe a typo : ReadDir vs Readdir, because ReadDir is further down the stack when you look through go's source code.

Reply:
No it was definitely ReadDir that I tried as I noticed the stdlib comment on Readdir originally which sparked this investigation https://pkg.go.dev/os#File.Readdir:

"Most clients are better served by the more efficient ReadDir method." What Go version are you on? Should upgrade to 1.20 on my end on my linux machine 😳

Reply:
Ah I am on go1.20 that's it

Reply:
I'm thinking maybe we should use an alternative approach here. The thing is, all the file names are well known ahead of time. The only thing that needs to be known is the list of hugetlb sizes (and for that, there's a function in cgroup1 package available (which needs to be wrapped in `once.Do`, but that's a different topic)).

When we know all page sizes, the code becomes something like this (an example from runc/libcontainer):

https://github.com/opencontainers/runc/blob/3a2c0c2565644d8a7e0f1dd594a060b21fa96cf1/libcontainer/cgroups/fs/hugetlb.go#L31-L59

Reply:
That does seem more ideal. I'll play with this tonight, thanks!

Reply:
As for the three versions of reading the file names,
- `os.Readdir` is the worst, because it calls `stat(2)` on every file;
- `os.ReadDir` is better as it doesn't do `stat` but instead wraps each entry into FileInfo interface;
- `os.Readdirnames` is the fastest as it just returns the list of names.

Since the name is everything we care about here, `os.Readdirnames` is the best.

Yet I would rewrite this using the approach without reading the directly (see above).

Reply:
I read wrong again :(